test/integration/sourcemap/solc_comparison.spec.ts

```diff
@@ -74,93 +74,94 @@ const samples = [
-                for (const node of unit.getChildren(true)) {
-                    const sourceInfo = node.sourceInfo;
+                [, sourceMap] = writeAST(writtenUnits, version);
+            });
 
-                    const solcStart = sourceInfo.offset;
-                    const solcLen = sourceInfo.length;
+            it("Source map from ASTWriter is compatible with source map from Solc", async () => {
+                for (const unit of writtenUnits) {
+                    for (const node of unit.getChildren(true)) {
+                        const sourceInfo = node.sourceInfo;
 
-                    if (!sourceMap.has(node)) {
-                        it(`Missing node ${node.type}#${node.id} must be an empty parameter list`, () => {
+                        const solcStart = sourceInfo.offset;
+                        const solcLen = sourceInfo.length;
+
+                        if (!sourceMap.has(node)) {
                             expect(node.type).toEqual("ParameterList");
                             expect((node as ParameterList).vParameters.length).toEqual(0);
-                        });
-
-                        continue;
-                    }
-
-                    const [compStart, compLen] = sourceMap.get(node) as [number, number];
-
-                    /**
-                     * Skip nodes without written source.
-                     * Usually these nodes are empty return `ParameterList`s.
-                     */
-                    if (solcLen === compLen && solcLen === 0) {
-                        continue;
-                    }
-
-                    const solcCoords = solcStart + ":" + solcLen;
-                    const compCoords = compStart + ":" + compLen;
-
-                    const solcFragment = getSourceFragment(solcStart, solcLen, writtenSource);
-                    const compFragment = getSourceFragment(compStart, compLen, writtenSource);
 
-                    if (compFragment === solcFragment && compCoords === solcCoords) {
-                        continue;
+                            continue;
+                        }
+
+                        const [compStart, compLen] = sourceMap.get(node) as [number, number];
+
+                        /**
+                         * Skip nodes without written source.
+                         * Usually these nodes are empty return `ParameterList`s.
+                         */
+                        if (solcLen === compLen && solcLen === 0) {
+                            continue;
+                        }
+
+                        const solcCoords = solcStart + ":" + solcLen;
+                        const compCoords = compStart + ":" + compLen;
+
+                        const solcFragment = getSourceFragment(solcStart, solcLen, writtenSource);
+                        const compFragment = getSourceFragment(compStart, compLen, writtenSource);
+
+                        if (compFragment === solcFragment && compCoords === solcCoords) {
```

Comment:
This is not a blocker, but the way this is written its not optimal. We only need `solcFragment` and `compFragment` if  `compCoords !== solcCoords`. So lines 123-131 can be optimized to:

```
if (solcStart === compStart && solcLen === compLen) {
    continue;
}

const solcCoords = solcStart + ":" + solcLen;
const compCoords = compStart + ":" + compLen;

const solcFragment = getSourceFragment(solcStart, solcLen, writtenSource);
const compFragment = getSourceFragment(compStart, compLen, writtenSource);
```